crates/router/src/core/revenue_recovery/types.rs

```diff
@@ -684,15 +690,24 @@ impl Action {
     }
 
     pub(crate) async fn decide_retry_failure_action(
+        state: &SessionState,
         db: &dyn StorageInterface,
         merchant_id: &id_type::MerchantId,
         pt: storage::ProcessTracker,
         revenue_recovery_payment_data: &storage::revenue_recovery::RevenueRecoveryPaymentData,
         payment_attempt: &payment_attempt::PaymentAttempt,
+        payment_intent: &PaymentIntent,
```

Comment:
Nit: There's inconsistency in imports, either fully import both payment attempt and payment intent, or qualify both. I'd prefer qualifying both.